src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/Utils.groovy

```diff
@@ -167,6 +162,27 @@ public class Utils {
         }
     }
 
+    /**
+     * Marks the containing stage with this name as a synthetic stage, with the appropriate {@link SyntheticContext}.
+     *
+     * @param stageName
+     * @param context
+     */
+    @Whitelisted
+    static void markSyntheticStage(String stageName, SyntheticContext context) {
```

Comment:
🐛  Better to use static final strings for the markers, i.e. "PRE" and "POST" or similar.   Creating an enum just for two values is kind of silly, and if we're just getting string values for 'em we might as well use that.